### PR TITLE
Update 2023.1.0

### PR DESCRIPTION
**Pull request by JeanChristopheMorinPerso**
Update our Intel compilers to 2023.1.0. We need to update them to be able to update dpctl to 0.14.2 (https://github.com/AnacondaRecipes/dpctl-feedstock/pull/3).

Additionally, after talking to Intel engineers, they suggested that we use `CORE-AVX512` instead of `COMMON-AVX512`. SO this PR updates the build flags too.

It's been tested against `dpctl` and `mkl_umath`.

### CODE REVIEW

File: abs.yaml [hunk 1]
```diff
@@ -3,3 +3,6 @@ build_parameters:
   - "--suppress-variables"
   - "--error-overlinking"
   - "--error-overdepending"
+
+channels:
```
Comment (cbouss) [inline abs.yaml]: Can you remove your channel?

**Review by cbouss — APPROVED** [review]
LGTM, thank you

**Review by sumit0190 — APPROVED** [review]
LGTM, but could you point to that discussion you had that suggested the use of `CORE-AVX512` over `COMMON-AVX512`? In the past, I've read conflicting information all over the web and some clarification would be nice, especially if it's coming straight from Intel.

**Comment by JeanChristopheMorinPerso** [comment]
>could you point to that discussion you had that suggested the use of CORE-AVX512 over COMMON-AVX512? In the past, I've read conflicting information all over the web and some clarification would be nice, especially if it's coming straight from Intel.

I'd be more than happy to point you to that information. The only problem is that it's an email exchange with an Intel engineer. Do you want me to forward the thread or would you prefer a more generic and open approach that would make this available to everybody?

**Comment by skupr-anaconda** [comment]
> > could you point to that discussion you had that suggested the use of CORE-AVX512 over COMMON-AVX512? In the past, I've read conflicting information all over the web and some clarification would be nice, especially if it's coming straight from Intel.
> 
> I'd be more than happy to point you to that information. The only problem is that it's an email exchange with an Intel engineer. Do you want me to forward the thread or would you prefer a more generic and open approach that would make this available to everybody?

In this docs is mentioned CORE-AVX512 https://crc.pitt.edu/sites/default/files/Intel%20Compilers%20Overview.pdf

And the difference is the next:
_SIMD feature set names For Intel® compilers. SIMD Feature Description_:
_**CORE-AVX512** May generate Intel® Advanced Vector Extensions 512 (Intel® AVX-512) Foundation instructions, Intel®
AVX-512 Conflict Detection instructions, **and other AVX-512 subsets which will be available on future Intel® XEON™ architecture**. Optimizes for Intel® processors that support Intel® AVX-512 instructions. Sets
–qopt-zmm-usage=low by default._

_**COMMON-AVX512** May generate Intel® Advanced Vector Extensions 512 (Intel® AVX-512) Foundation instructions and Intel® AVX-512 Conflict Detection instructions. Optimizes for Intel® processors that support Intel® AVX512 instructions. Sets –qopt-zmm-usage=high by default._

The source is https://events.prace-ri.eu/event/718/attachments/693/993/ato_handout.pdf

So **CORE-AVX512** is a better choice

**Review by skupr-anaconda — APPROVED** [review]

**Comment by JeanChristopheMorinPerso** [comment]
@skupr-anaconda actually the real sourc of these descriptions is https://www.intel.com/content/www/us/en/docs/cpp-compiler/developer-guide-reference/2021-8/ax-qax.html.

**Comment by JeanChristopheMorinPerso** [comment]
From Intel (oleksandr.pavlyk@intel.com):

>Per Intel LLVM compiler 2023.1.0 documentation for “[-x](https://www.intel.com/content/www/us/en/docs/dpcpp-cpp-compiler/developer-guide-reference/2023-1/x-qx.html)” and “[-ax](https://www.intel.com/content/www/us/en/docs/dpcpp-cpp-compiler/developer-guide-reference/2023-1/ax-qax.html#GUID-2D881A91-C5D7-4DDD-84B1-FB9D0D597F4D)” flags, these settings remain valid.
>>   -xSSE4.2  - specifies minimal required HW capability. An attempt to execute binary on hardware that does not feature SSE4.2 dumps core with Illegal instruction message.
>>   -ax  flag indicates list of alternative architectural targets for which compiler generates code where deemed appropriate with dynamic dispatching.
>
>Extension import failure my arise due to symbol resolution issue (if so, ctypes.cdll.LoadLibrary should also fail to load a native extension),
>or it may occur due to illegal instruction encountered during Python initialization function execution. It is useful to acquire output of “lscpu” of the machine where such a problem manifests itself.
>
>Using “-xSSE4.2 -axCORE-AVX2,COMMON-AVX512” for certain portions of the code the compiler may produce 3 binary translations: one using only SSE4.2 instructions, second using instructions permitted by CORE-AVX2, and third instructions permitted by COMMON-AVX512.
>
>The latter, per documentation, may also include instructions from CORE-AVX2 extension set, but it can be guaranteed to be executable on AVX2-only machines, since compiler is free to also use instructions outside of CORE-AVX2 extension set.
>Hence, the change that helped effectively turned off AVX2 instructions.

(There is a typo in "but it can be guaranteed to be executable" an it should instead use "can't" instead of "can").

From the same person:
>While this is pending, it occurs to me that COMMON-AVX512 on the line https://github.com/AnacondaRecipes/intel-compilers-repack-feedstock/blob/master/recipe/update-activate-and-repack.sh#L14 and similar is best replaced with CORE-AVX512 (wider target extension set), so I would advise doing “-xSSE4.2 -axCORE-AVX2,CORE-AVX512”.

**Pull request merged**